## Importing Data Set on Aviation Risk

In [19]:
import pandas as pd

## Loading and read the CSV file

In [20]:
df=pd.read_csv("../Phase1/data/AviationData.csv", encoding = "ISO-8859-1")

C:\Users\NCG\AppData\Local\Temp\ipykernel_5820\2280785669.py:1: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv("../Phase1/data/AviationData.csv", encoding = "ISO-8859-1")


## Using .info() and .describe() to view missing values and statistical measures

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [22]:
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

## Using .head(10) to view the first 10 incidents as a dataframe

In [23]:
df.head(10)

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   
5  20170710X52551           Accident      NYC79AA106  1979-09-17   
6  20001218X45446           Accident      CHI81LA106  1981-08-01   
7  20020909X01562           Accident      SEA82DA022  1982-01-01   
8  20020909X01561           Accident      NYC82DA015  1982-01-01   
9  20020909X01560           Accident      MIA82DA029  1982-01-01   

           Location        Country   Latitude  Longitude Airport.Code  \
0   MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1    BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2     Saltville, VA  United States  36.922223 -81.878056          NaN   
3        EUREKA, CA  United States        NaN        NaN          NaN   
4        Canton, OH  United States        NaN        NaN          NaN   
5        BOSTON, MA  United States  42.445277 -70.758333          NaN   
6        COTTON, MN  United States        NaN        NaN          NaN   
7       PULLMAN, WA  United States        NaN        NaN          NaN   
8  EAST HANOVER, NJ  United States        NaN        NaN          N58   
9  JACKSONVILLE, FL  United States        NaN        NaN          JAX   

         Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0                 NaN  ...          Personal         NaN                  2.0   
1                 NaN  ...          Personal         NaN                  4.0   
2                 NaN  ...          Personal         NaN                  3.0   
3                 NaN  ...          Personal         NaN                  2.0   
4                 NaN  ...          Personal         NaN                  1.0   
5                 NaN  ...               NaN  Air Canada                  NaN   
6                 NaN  ...          Personal         NaN                  4.0   
7  BLACKBURN AG STRIP  ...          Personal         NaN                  0.0   
8             HANOVER  ...          Business         NaN                  0.0   
9   JACKSONVILLE INTL  ...          Personal         NaN                  0.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   
5                    NaN                  1.0            44.0   
6                    0.0                  0.0             0.0   
7                    0.0                  0.0             2.0   
8                    0.0                  0.0             2.0   
9                    0.0                  3.0             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  
5               VMC                  Climb  Probable Cause       19-09-2017  
6               IMC                Unknown  Probable Cause       06-11-2001  
7               VMC                Takeoff  Probable Cause       01-01-1982  
8               IMC                Landing  Probable Cause       01-01-1982  
9               IMC                 Cruise  Probable Cause       01-

## Viewing Missing Data in "Weather.Condition"

In [24]:
df['Weather.Condition'].isna().value_counts()

False    84397
True      4492
Name: Weather.Condition, dtype: int64

4492 missing weather conditions in our data set

## Filtering for Weather Condition and Injury Types

In [27]:
df[['Weather.Condition', 'Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries']]

Weather.Condition  Total.Fatal.Injuries  Total.Serious.Injuries  \
0                   UNK                   2.0                     0.0   
1                   UNK                   4.0                     0.0   
2                   IMC                   3.0                     NaN   
3                   IMC                   2.0                     0.0   
4                   VMC                   1.0                     2.0   
...                 ...                   ...                     ...   
88884               NaN                   0.0                     1.0   
88885               NaN                   0.0                     0.0   
88886               VMC                   0.0                     0.0   
88887               NaN                   0.0                     0.0   
88888               NaN                   0.0                     1.0   

       Total.Minor.Injuries  
0                       0.0  
1                       0.0  
2                       NaN  
3                       0.0  
4                       NaN  
...                     ...  
88884                   0.0  
88885                   0.0  
88886                   0.0  
88887                   0.0  
88888                   0.0  

[88889 rows x 4 columns]